src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -682,7 +829,78 @@ public void CopyTo(Array array, int index)
                     throw new ArgumentException(SR.Argument_InvalidOffLen);
                 }
 
-                for (int i = 0; i < m_length; i++)
+                int i = 0;
+
+                if (m_length < 32)
+                    goto LessThan32;
+
+                if (Avx2.IsSupported)
+                {
+                    Vector256<byte> shuffleMask;
+                    Vector256<byte> bitMask;
+                    Vector256<byte> ones = Vector256.Create((byte)1);
+                    fixed (byte* ptr = ShuffleMask_CopyToBoolArray)
```

Comment:
It's a `ROSpan<byte>` which the C# compiler emits as an ecma-335 data declaration (which is embedded directly in the PE header file) and loads into the returned `ROSpan<byte>` as a raw pointer. It should be safe to just `byte* ptr = Unsafe.AsPointer(ref MemoryMarshal.Getreference(Mask))` as the data is embedded and accessed directly from the PE header and should be immovable.

CC. @jkotas, could you confirm that the GC doesn't track/relocate "data declarations" (and whether assuming that this particular code pattern backs pinned data is "safe")?
-- Noting that this would only apply to the `ROSpan<byte>` and `ROSPan<sbyte>` patterns, since other primitive types go through a JIT helper and have an array allocated for them.

Reply:
> could you confirm that the GC doesn't track/relocate "data declarations"

It does not, but I do not think it is a good idea to be promoting tricky code like this.

The most natural way to do this is using `static readonly Vector256<byte>`. I understand that the JIT emits a few extra instructions for this today. We should fix that if it matters.